File: 백준/Silver/14248. 점프 점프/점프 점프.cc
#include <bits/stdc++.h>

using namespace std;

int main() {
    ios_base::sync_with_stdio(false);cin.tie(NULL);cout.tie(NULL);
    int n, s, count = 0;
    cin >> n;
    vector<int> a(n);
    vector<bool> visit(n);
    for (int i = 0; i < n; i++)cin >> a[i];
    cin >> s;
    stack<int> dfs;
    dfs.push(s - 1);
    visit[s - 1] = true;
    
    while (!dfs.empty()) {
        int i = dfs.top();
        dfs.pop();
        count++;
        int left = i - a[i], right = i + a[i];
        if (left >= 0 && !visit[left]) {
            visit[left] = true;
            dfs.push(i - a[i]);
        }
        if (right < n && !visit[right]) {
            visit[right] = true;

            dfs.push(right);
        }
    }
    cout << count;
}

File: 백준/Silver/14248. 점프 점프/점프 점프.java
import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int count = 0;
        
        int[] a = new int[n];
        boolean[] visit = new boolean[n];
        
        for (int i = 0; i < n; i++) {
            a[i] = sc.nextInt();
        }
        
        int s = sc.nextInt();  // 시작 지점 입력
        
        Stack<Integer> dfs = new Stack<>();
        dfs.push(s - 1);
        visit[s - 1] = true;
        
        while (!dfs.isEmpty()) {
            int i = dfs.pop();
            count++;
            int left = i - a[i];
            int right = i + a[i];
            
            if (left >= 0 && !visit[left]) {
                visit[left] = true;
                dfs.push(left);
            }
            if (right < n && !visit[right]) {
                visit[right] = true;
                dfs.push(right);
            }
        }
        
        System.out.println(count);
    }
}
